index.js

```diff
@@ -60,11 +59,13 @@ require("dotenv").config();
 
 app.use(express.json());
 app.use(cors());
+app.use(jwtMiddleware.unless({ path: ["/"] }));
+app.use(currentUserCheck);
```

Comment:
I've installed express-unless directly to use the unless function. I think I can also get away with simply writing an if statement within currentUserCheck to look for the req.path if its the root path, then move past that middleware.

I also added a check within currentUserCheck that checks whenther there is an auth header to run the middleware.

Reply:
I see you added a line to skip `currentUserCheck` if there's an auth header, but I feel like we should also use `express-unless` there too for consistency. It feels to me like `currentUserCheck` should only run on the same endpoints that `jwtMiddleware` runs because then that means the user is logged in, you know?

I was thinking we can maybe make a whitelist of routes that don't need auth and share that array between the two middlewares. I guess if there somehow isn't an auth header the server might crash, so maybe we can also throw a 401 unauthorized error in `currentUserCheck` if `req.auth` is undefined. What do you think?